xrpl/transaction/account_set.go

```diff
@@ -123,37 +124,85 @@ func (s *AccountSet) Flatten() FlatTransaction {
 	if s.ClearFlag != 0 {
 		flattened["ClearFlag"] = s.ClearFlag
 	}
-	if s.Domain != "" {
-		flattened["Domain"] = s.Domain
+	if s.Domain != nil {
+		flattened["Domain"] = *s.Domain
 	}
-	if s.EmailHash != "" {
+	if s.EmailHash != nil {
 		flattened["EmailHash"] = s.EmailHash.String()
 	}
-	if s.MessageKey != "" {
-		flattened["MessageKey"] = s.MessageKey
+	if s.MessageKey != nil {
+		flattened["MessageKey"] = *s.MessageKey
 	}
-	if s.NFTokenMinter != "" {
-		flattened["NFTokenMinter"] = s.NFTokenMinter
+	if s.NFTokenMinter != nil {
+		flattened["NFTokenMinter"] = *s.NFTokenMinter
 	}
 	if s.SetFlag != 0 {
 		flattened["SetFlag"] = s.SetFlag
 	}
-	if s.TransferRate != 0 {
-		flattened["TransferRate"] = s.TransferRate
+	if s.TransferRate != nil {
+		flattened["TransferRate"] = *s.TransferRate
 	}
-	if s.TickSize != 0 {
-		flattened["TickSize"] = s.TickSize
+	if s.TickSize != nil {
+		flattened["TickSize"] = *s.TickSize
 	}
-	if s.WalletLocator != "" {
+	if s.WalletLocator != nil {
 		flattened["WalletLocator"] = s.WalletLocator.String()
 	}
-	if s.WalletSize != 0 {
-		flattened["WalletSize"] = s.WalletSize
+	if s.WalletSize != nil {
+		flattened["WalletSize"] = *s.WalletSize
 	}
 
 	return flattened
 }
 
+// The domain that owns this account, as a string of hex representing the.
+// ASCII for the domain in lowercase.
+func Domain(value string) *string {
```

Comment:
Nice, we should move it to a `types` subpackage on the same folder

Reply:
Done, I moved them to their own files in `types`.